### ISSUE TITLE
Bug S1: addJobTitles search producing strange results 

### ISSUE THREAD

**Issue by joshdanielson** [loconomics/loconomics#651]
- only showing 5 job titles that are completely unrelated (see screenshot). Same for any other job title.
- If submitting "Photographer" (from the typed text), it created a listing for "Event Planner"
![screen shot 2017-09-06 at 3 49 44 pm](https://user-images.githubusercontent.com/1202838/30141399-1cfb815a-9330-11e7-802a-a7e28ceeff11.png)

**Comment by IagoSRL** [comment]
The same search at home page gives different results; not sure if must use the same underlying search, will check

**Comment by IagoSRL** [comment]
I found and fixed the UI part of this bug (showing up the correct autocomplete results). But still, if choosing 'add new' using the 'photographer' text, the back-end founds an Event Planner as closest result. Maybe is because it contains the 'photographer' keyword, but still results from search operations should match the same at front-end and back-end.

**Comment by IagoSRL** [comment]
Fixed the back-end part, was using the old lookup for job titles matching a name, I removed that (and a REST API endpoint using it, /job-titles/autocomplete, not in use for a while and replaced with the /search API) and used the same internal method that the search API used at the UI, so both ways matches the same value.

I pushed this to *dev* and is528 branch.

Note that the home page autocomplete is the new accessible one (still to be applied to other parts and further testing). Because of some shared styles changes, the results at add-job-title may look not well, I'm replacing that with the new autocomplete.
